tests/unit/test_compress.py

```diff
@@ -0,0 +1,276 @@
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import gzip
+import io
+
+import pytest
+
+from botocore.compress import maybe_compress_request
+from botocore.config import Config
+from tests import mock
+
+
+def _op_with_compression():
+    op = mock.Mock()
+    op.request_compression = {'encodings': ['gzip']}
+    op.has_streaming_input = False
+    return op
+
+
+def _op_with_multiple_compressions():
+    op = mock.Mock()
+    # TODO: Update second value to a different encoding. Currently only gzip
+    # is supported.
+    op.request_compression = {'encodings': ['gzip', 'gzip']}
```

Comment:
This test case is to ensure that compression doesn't run twice even when more than one encoding is available so we need real encodings that have existing methods in the SDK to properly validate this. Since `gzip` is currently the only one, we have to use it twice.

Reply:
> so we need real encodings that have existing methods in the SDK to properly validate this.

What's prompting that requirement? If you configure this as `["gzip", "foo"]`, you're testing that only the first is used. If we reach foo, we've failed your test case. If they're both `gzip` and you assert `gzip` was used, we're adding ambiguity on which was selected.

Reply:
`_assert_compression` will only ever decompress once so if it did use gzip twice the test would fail. To your point though, there is still ambiguity on which gzip in the list was selected. However, `foo` will never compress since its not a supported encoding. How do we validate that `foo` was never reached? I don't really see a clear way of doing that.

Reply:
On second thought, I'll change to a mocked value like "foo". Since multiple compressions aren't supported yet, we don't necessarily need to test for this behavior, just that compression is only run once. If and when we do, we can add a test case that explicitly tests an operation with multiple supported compression encodings.

Reply:
I think there's a lot of ways we can ensure this is checked. A simple option would be to do something like:

```
def bad_compression():
    raise ValueError("Reached unintended compression algorithm 'foo'")
    
MOCK_COMPRESSION = {"foo": bad_compression}.update(COMPRESSION_MAPPING)
with mock.patch('botocore.compress.COMPRESSION_MAPPING', MOCK_COMPRESSION):

Reply:
Thanks! I guess I was worried about messing with internals and using mocking, but this seems like a decent option.